modelling/src/main/java/org/axonframework/modelling/configuration/ModellingConfigurer.java

```diff
@@ -0,0 +1,115 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.modelling.configuration;
+
+import jakarta.annotation.Nonnull;
+import org.axonframework.configuration.ApplicationConfigurer;
+import org.axonframework.configuration.AxonApplication;
+import org.axonframework.configuration.Component;
+import org.axonframework.configuration.ComponentDecorator;
+import org.axonframework.configuration.ConfigurationEnhancer;
+import org.axonframework.configuration.DelegatingConfigurer;
+import org.axonframework.configuration.MessagingConfigurer;
+import org.axonframework.configuration.Module;
+import org.axonframework.configuration.ModuleBuilder;
+import org.axonframework.configuration.NewConfigurer;
+
+import java.util.function.Consumer;
+
+/**
+ * The modelling {@link NewConfigurer} of Axon Framework's configuration API, providing registration methods to, for
+ * example, register a {@link StatefulCommandHandlingModule}.
+ * <p>
+ * This configurer does not set any defaults other than the defaults granted by the {@link MessagingConfigurer} it
+ * wraps.
+ *
+ * @author Steven van Beelen
+ * @since 5.0.0
+ */
+public class ModellingConfigurer
+        extends DelegatingConfigurer<ModellingConfigurer>
+        implements ApplicationConfigurer<ModellingConfigurer> {
+
+    /**
+     * Build a default {@code ModellingConfigurer} instance with several modelling defaults.
+     * <p>
+     * Besides the specific operations, the {@code ModellingConfigurer} allows for configuring generic
+     * {@link Component components}, {@link ComponentDecorator component decorators},
+     * {@link ConfigurationEnhancer enhancers}, and {@link Module modules} for an application using entity modelling.
+     * <p>
+     * Note that this configurer uses a {@link MessagingConfigurer} to support all this in a message-driven style.
+     *
+     * @return A {@code ModellingConfigurer} instance for further configuring.
+     */
+    public static ModellingConfigurer create() {
+        return new ModellingConfigurer(MessagingConfigurer.create());
+    }
+
+    /**
+     * Construct a {@code ModellingConfigurer} using the given {@code delegate} to delegate all registry-specific
+     * operations to.
+     * <p>
+     * It is recommended to use the {@link #create()} method in most cases instead of this constructor.
+     *
+     * @param delegate The delegate {@code MessagingConfigurer} the {@code ModellingConfigurer} is based on.
+     */
+    public ModellingConfigurer(@Nonnull MessagingConfigurer delegate) {
+        super(delegate);
+    }
+
+    /**
+     * Registers the given stateful command handling {@code moduleBuilder} to use in this configuration.
```

Comment:
What do you think of something like this?
```suggestion
     * Registers the given {@link ModuleBuilder builder} for a {@link StatefulCommandHandlingModule} to use in this configuration.
```